quill-engine/src/main/scala/io/getquill/idiom/StatementInterpolator.scala

```diff
@@ -43,19 +46,15 @@ object StatementInterpolator {
       case lift: Lift => liftTokenizer.token(lift)
     }
 
-  implicit def tagTokenizer: Tokenizer[Tag] =
+  implicit val tagTokenizer: Tokenizer[Tag] =
     Tokenizer[Tag] {
       case tag: ScalarTag    => ScalarTagToken(tag)
       case tag: QuotationTag => QuotationTagToken(tag)
     }
 
-  implicit def liftTokenizer: Tokenizer[Lift] =
+  implicit val liftTokenizer: Tokenizer[Lift] =
     Tokenizer[Lift] {
-      case tag: ScalarTag    => ScalarTagToken(tag)
-      case tag: QuotationTag => QuotationTagToken(tag)
-      case lift: ScalarLift  => ScalarLiftToken(lift)
-      // TODO Longer Explanation
-      case lift: Tag => fail("Cannot tokenizer a non-scalar tagging.")
```

Comment:
AFAICT, a `Lift` can't be a `ScalarTag`, nor a `QuotationTag` nor a `Tag`

Reply:
Good catch! I think there was a point at which Tag was a sub-type of Lift but I changed that a while ago.